component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java

```diff
@@ -104,6 +104,13 @@ public String toString() {
         }
     }
 
+    @Override
+    public Builder withNewSchema(final Schema schema) {
+        final BuilderImpl builder = new BuilderImpl(schema);
+        schema.getEntriesOrdered().forEach(e -> builder.with(e, values.get(e.getName())));
```

Comment:
`schema.getAllEntries` better candidate (no unuseful ordered, meta data into account)
filter with existing field ? Then get value from "this"
```java
 public Builder withNewSchema(final Schema newSchema) {
        final AvroRecordBuilder builder = new AvroRecordBuilder(newSchema);
        newSchema.getAllEntries()
            .filter(e -> Objects.equals(this.schema.getEntry(e.getName()) , e)) // ensure entry are of same type (avoid put a Long into a String field for example.
            .forEach(e -> {
                      Object value = RecordImpl.this.get(Object.class, e.getName());
                      if (value != null) {
                         builder.with(e, value);
                      }
               });
```

Reply:
Right for using getAllEntries and ensuring same entries type.
Will not filter on null as it may be intended...